components/brave_wallet_ui/common/reducers/wallet_reducer.ts

```diff
@@ -300,6 +301,10 @@ reducer.on(WalletActions.unapprovedTxUpdated, (state: any, payload: UnapprovedTx
 })
 
 reducer.on(WalletActions.transactionStatusChanged, (state: WalletState, payload: TransactionStatusChanged) => {
+  if (isHardwareAccount(state.accounts, payload.txInfo.fromAddress)) {
+    return state
```

Comment:
why are we early return here?

Reply:
we set hardware transaction status as approved too early, moved it now to the last step before publishing and removed this code.